### PR TITLE
Facet logic now AND between facets OR within facets

### PR DESCRIPTION
**Pull request by Tehsurfer**
# Description
Currently, multiple facets in a search are combined using AND boolean operators. 

This PR switches the facet search to combining multiple facets like the image below. AND between facets, OR inside facets.

This was suggested feedback from Gazelle on the sidebar.

Can be viewed at:
https://jesse-api-test.herokuapp.com/maps

image

## Type of change

Delete those that don't apply.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

Tested with mapintegratedvuer and the included python tests

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added unit tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by Tehsurfer** [comment]
Updated description to include live demo

File: app/process_kb_results.py [hunk 1]
```diff
@@ -74,32 +74,57 @@ def create_filter_request(query, terms, facets, size, start):
...
-    if query is not '':
-        data['query']['bool']['must'] = {
-          "query_string": {
-            "query": f"{query}",
-            "default_operator": "and",
-            "lenient": "true",
-            "type": "best_fields"
-          }
-        }
+    print(data)
```
Comment (alan-wu) [inline app/process_kb_results.py:86]: Remove this please.

Comment (Tehsurfer) [reply app/process_kb_results.py]: Removed

**Comment by alan-wu** [comment]
Looks good to me.

**Comment by alan-wu** [comment]
Looks good to me.

**Review by nickerso — APPROVED** [review]
reviewed by @alan-wu

**Pull request merged**